jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -1735,22 +1735,64 @@ static MethodHandle filterArgument(MethodHandle target, int pos, MethodHandle fi
 		return Lookup.PUBLIC_LOOKUP;
 	}
 	
-	/*[IF Sidecar19-SE]*/
+	/*[IF Sidecar19-SE-B175]*/
 	/**
-	 * Return a MethodHandles.Lookup object that is only able to access <code>private</code> members.
+	 * Return a MethodHandles.Lookup object with full capabilities including the access 
+	 * to the <code>private</code> members in the requested class
 	 * 
 	 * @param targetClass - the requested class containing private members
 	 * @param callerLookup - a Lookup object specific to the caller
 	 * @return a MethodHandles.Lookup object with private access to the requested class
+	 * @throws NullPointerException - if any of the arguments is null
 	 * @throws IllegalArgumentException - if the requested Class is a primitive type or an array class
-	 * @throws NullPointerException - if any of the arguments are null
 	 * @throws IllegalAccessException - if access checking fails
 	 * @throws SecurityException - if the SecurityManager prevents access
 	 */
-	public static MethodHandles.Lookup privateLookupIn(Class<?> targetClass, MethodHandles.Lookup callerLookup) throws IllegalArgumentException, NullPointerException, IllegalAccessException, SecurityException {
-		throw new UnsupportedOperationException("The method has not yet been implemented for now"); //$NON-NLS-1$
+	public static MethodHandles.Lookup privateLookupIn(Class<?> targetClass, MethodHandles.Lookup callerLookup) throws NullPointerException, IllegalArgumentException, IllegalAccessException, SecurityException {
+		if (Objects.isNull(targetClass) || Objects.isNull(callerLookup)) {
+			/*[MSG "K065S", "Both the requested class and the caller lookup must be non-null"]*/
+			throw new NullPointerException(com.ibm.oti.util.Msg.getString("K065S")); //$NON-NLS-1$
+		}
+		
+		if (targetClass.isPrimitive() || targetClass.isArray()) {
+			/*[MSG "K065T", "The target class: {0} must not be a primitive type or an array class"]*/
+			throw new IllegalArgumentException(com.ibm.oti.util.Msg.getString("K065T", targetClass.getCanonicalName())); //$NON-NLS-1$
+		}
+		
+		Module targetClassModule = targetClass.getModule();
+		String targetClassPackageName = targetClass.getPackageName();
+		Module accessClassModule = callerLookup.accessClass.getModule();
+		
+		/* Check whether the named module containing the old lookup can read the module containing the target class.
+		 * Note: an unnamed module can read any module.
+		 */
+		if (accessClassModule.isNamed() && (!accessClassModule.canRead(targetClassModule))) {
```

Comment:
`accessClassModule.isNamed()` is redundant. `Module.canRead()` handles unnamed module case internally.

Reply:
Agreed and fixed.